tests/test__mongomock.py

```diff
@@ -3062,6 +3062,63 @@ def test__aggregate_cond_mongodb_to_bool(self):
             }},
         ])
 
+    def test__aggregate_concatArrays(self):
+        self.cmp.do.drop()
+        self.cmp.do.insert_one({
+            '_id': 1,
+            'a': [1, 2],
+            'b': ['foo', 'bar', 'baz'],
+            'c': {
+                'arr1': [123]
+            }
+        })
+        pipeline = [{
+            '$project': {
+                '_id': 0,
+                'concat': {'$concatArrays': ['$a', ['#', '*'], '$c.arr1', '$b']},
+                'concat_array_expression': {'$concatArrays': '$b'},
+                'concat_tuples': {'$concatArrays': ((1, 2, 3), (1,))},
+                'concat_none': {'$concatArrays': None},
+                'concat_missing_field': {'$concatArrays': '$foo'},
+                'concat_none_item': {'$concatArrays': ['$a', None, '$b']},
+                'concat_missing_field_item': {'$concatArrays': [[1, 2, 3], '$c.arr2']}
+            }
+        }]
+        self.cmp.compare.aggregate(pipeline)
+
+    def test__aggregate_concatArrays_exceptions(self):
+        self.cmp.do.drop()
+        self.cmp.do.insert_one({
+            '_id': 1,
+            'a': {
+                'arr1': [123]
+            }
+        })
+
+        pipeline_parameter_not_array = [
+            {
+                '$project': {
+                    'concat_parameter_not_array': {'$concatArrays': 42}
+                }
+            }
+        ]
+        with self.assertRaises(OperationFailure):
```

Comment:
Please use the `self.cmp.compare_exceptions.aggregate(pipeline_parameter_not_array)` function, same below. This ensures we use the same type of error in Pymongo and in mongomock without the need of the extra code.